### ISSUE TITLE
Only Laneline

### ISSUE THREAD

**Issue by PeggyPeppa** [ybarancan/STSU#10]
Hello, how can I change the code if I only want to implement the lane line part? Setting args.objects  to false seems not work

**Comment by ybarancan** [comment]
Hi,
I changed some files. Specifically, dataset files are changed to return obj_exists = False if args.objects = True and there are some small changes in other files as well. Let me know if you run into any problems.
Best

**Comment by PeggyPeppa** [comment]
Thank you. I will try to fix this problem.

**Comment by MintGreen-hub** [comment]
Hi ~ We use one picture to train this model."token:8ed2a7ea9a894eac8b3cd43b706f39bc; epoch:2000; lr=5e-4; clip_max_norm=5" and use the pretrained model "https://data.vision.ee.ethz.ch/cany/STSU/transformer.pth“ to start training. But the results is incomprehensible. The first and second graphs represent our truth values and training results, respectively. 
![batch_0_gt_visible_all_roads](https://user-images.githubusercontent.com/60730828/198045027-e00d2227-08a5-4a0
![all_matched_road_16](https://user-images.githubusercontent.com/60730828/198045055-8eb4986b-daeb-4259-8bdd-1c071fea6c33.jpg)
d-9cd4-feabdc25da91.jpg)

Also for the this line showed in the third and fourth picture, it is interrupted by vis_mask. Dose this kind of line affect training?
![gt_road_6](https://user-images.githubusercontent.com/60730828/198045112-3cead49c-7c66-4db1-8851-6788f24f8111.jpg)
![gt_road_18](https://user-images.githubusercontent.com/60730828/198045126-08c65ef3-dace-4a86-8c8d-75de74730ff3.jpg)

What's your opinion about this two problems?

**Comment by PeggyPeppa** [comment]
Hi ~ We use one picture to train this model."token:8ed2a7ea9a894eac8b3cd43b706f39bc; epoch:2000; lr=5e-4; clip_max_norm=5" and use the pretrained model "[https://data.vision.ee.ethz.ch/cany/STSU/transformer.pth“](https://data.vision.ee.ethz.ch/cany/STSU/transformer.pth%E2%80%9C) to start training. But the results is incomprehensible. The first and second graphs represent our truth values and training results, respectively.

![batch_0_gt_visible_all_roads](https://user-images.githubusercontent.com/60730828/198045437-26571ce5-1b00-4e30-b67d-bcb7213dfcc9.jpg)

![all_matched_road_16](https://user-images.githubusercontent.com/60730828/198045471-7e3a0cba-93f3-4c9a-a77d-a8cdfb9e4756.jpg)

Also for the this line showed in the third and fourth picture, it is interrupted by vis_mask. Dose this kind of line affect training?

![gt_road_6](https://user-images.githubusercontent.com/60730828/198045517-d9297a13-1d8b-4339-b8ec-e11d8dae0924.jpg)

![gt_road_18](https://user-images.githubusercontent.com/60730828/198045553-9262b5a7-1966-4ec7-8f12-8dabe0c06984.jpg)




What's your opinion about this two problems?

**Comment by PeggyPeppa** [comment]
Besides when we use Cityscapes pretrained Deeplab model to start our training. we can't get the desired result.

**Comment by ybarancan** [comment]
The results you show are raw unmerged estimates. With graph connectivity estimates, a post-processing step merges the individual centerlines. Apart from that, the results seem normal. Can you give the numerical results as well? As for the Deeplab training, it takes some epochs (~100) to reach the reported results. You can try playing with learning rate to reach a desired performance faster. And you should keep monitoing the validation scores to get the best model.

**Comment by PeggyPeppa** [comment]
Thanks a lot! But in my opinion when we use one picture to train model, it may cause overfitting which means the prediction of this picture will be very close to the groundtruth after many iterations. But the result seems not fit this conjecture. Or there are some tricks in the model to overcome overfitting?

**Comment by ybarancan** [comment]
There is dropout and weight decay. Also, the backbone is not trained. If you started from the pre-trained network, that might also act as a regularizer. But in my training of the network, I observed some overfitting but only after 100 epochs. So I would assume if you keep training with the same image, it would overfit. You can also make sure that, you are indeed training with one image. It might be that, by some mistake, the network is being trained with the whole dataset.

**Comment by PeggyPeppa** [comment]
Thanks!
